### PR TITLE
Feature/testing doc

### PR DESCRIPTION
**Pull request by harrisj**
Here is a first cut at a TEST.md file. Note that I also had to update a gem since the latest version of PhantomJS has changed. Finally, I added a note to a test that will fail if the MySQL is not configured correctly

### CODE REVIEW

File: TEST.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+MyUsa uses several different tools for testing various aspects of its operation:
+
+* [Rspec](http://rspec.info/), a behavior-driven development testing framework for Ruby
+* [Capybara](http://jnicklas.github.io/capybara/), a library for integration testing
+* [Code Climate](https://codeclimate.com/), a service for automated code reviews
+
+These testing mechanisms uses some additional tools to get what they need done
+
+* [Site Prism](https://github.com/natritmeyer/site_prism), a simple DSL for the DOM within Capybara
```
Comment (yozlet) [inline TEST.md]: More than that - it's an implementation of the Page Object Model (see http://www.guru99.com/page-object-model-pom-page-factory-in-selenium-ultimate-guide.html ), which is fundamental to how we write our feature tests.

File: TEST.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# How to Setup and Run Tests
+
+MyUsa uses several different tools for testing various aspects of its operation:
+
+* [Rspec](http://rspec.info/), a behavior-driven development testing framework for Ruby
+* [Capybara](http://jnicklas.github.io/capybara/), a library for integration testing
+* [Code Climate](https://codeclimate.com/), a service for automated code reviews
+
+These testing mechanisms uses some additional tools to get what they need done
```
Comment (adelevie) [inline TEST.md]: uses -> use

File: TEST.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# How to Setup and Run Tests
+
+MyUsa uses several different tools for testing various aspects of its operation:
```
Comment (adelevie) [inline TEST.md]: MyUsa -> MyUSA

**Comment by adelevie** [comment]
The doc looks great. I found and marked down two typos that don't really matter.

**Pull request merged**